cmd/subnetcmd/deploy.go

```diff
@@ -260,16 +283,23 @@ func deploySubnet(cmd *cobra.Command, args []string) error {
 
 	ux.Logger.PrintToUser("Your Subnet's control keys: %s", controlKeys)
 
-	// prompt for threshold
+	// validate and prompt for threshold
+	if int(threshold) > len(controlKeys) {
+		return fmt.Errorf("given threshold is greater than number of control keys")
+	}
 	if len(controlKeys) > 0 && threshold == 0 {
-		threshold, err = getThreshold(len(controlKeys))
-		if err != nil {
-			return err
+		if len(controlKeys) == 1 {
```

Comment:
Nit: It looks more elegant to me to do this inside `getThreshold`, so we could then only do `threshold,err = getThreshold()`.

Up to you